### ISSUE TITLE
Undeprecate it by using MGL for opengl on macos?

### ISSUE THREAD

**Issue by alesya-h** [rumblesan/improviz#73]
Hi. How feasible would it be to switch the project to https://github.com/openglonmetal/MGL ? I'm a Linux user, but it's a shame this was deprecated simply because apple decided it doesn't like opengl anymore. MGL seems maintained enough

**Comment by rumblesan** [comment]
Hey! Thanks for the heads up on the OpenGL on Metal project, I wasn't aware of it. The question of how feasible it might be to port to it is maybe a bit difficult to cleanly answer though.
I think the main thing would be changing the haskell OpenGL and OpenGLRaw libraries to use MGL. If that was done (and MGL supported the features that improviz uses) then it could maybe just be a question of bumping the library and things would work. My suspicion is that updating those libraries would be non-trivial unfortunately and that's kind of the rub with this project as a whole.

Most of the work to fix it on MacOs now would be non-trivial. Either updating those libraries, switching to using a different graphics layer library (there are haskell bindings for vulkan https://hackage.haskell.org/package/vulkan) rewriting a graphics layer in a language better suited to it and then binding to that (OpenGL in haskell is kind of miserable to be honest) or even just rewriting the whole thing. And unfortunately I don't really have the energy, inclination, or in some ways the ability to do that kind of deep software engineering work any more (burnout is rough).

TLDR; It's almost certainly possible, but probably not by me in the foreseeable future.
